src/machine.rs

```diff
@@ -351,7 +351,8 @@ pub struct MiriMachine<'mir, 'tcx> {
     pub tcx: TyCtxt<'tcx>,
 
     /// Stacked Borrows global data.
-    pub stacked_borrows: Option<stacked_borrows::GlobalState>,
+    pub borrow_tracker_method: BorrowTrackerMethod,
+    pub borrow_tracker_data: Option<borrow_tracker::GlobalState>,
```

Comment:
`borrow_tracker_method` should not be needed, since it is entirely determined by `borrow_tracker_data`. Then you can rename that field to `borrow_tracker`.

Reply:
Oh I see, the global borrow_tracker_data is the same no matter the borrow tracking module.

But still, we should only have one field for the borrow tracker here. `new_allocation` should become a function on `borrow_tracker::GlobalState` which then has to internally remember whatever is needed.